packages/cli/src/cmds/agentInstaller/agentProcedure.ts

```diff
@@ -108,6 +106,151 @@ export default abstract class AgentProcedure {
     return validationResult;
   }
 
+  async filesLocallyModified(): Promise<string[]> {
+    let files: string[] = [];
+
+    let gitStatus;
+    try {
+      const stdout = runSync(new CommandStruct('git', ['status', '-s'], this.installer.path));
+      if (stdout.length > 0) {
+        const lines = stdout.split('\n');
```

Comment:
You may enjoy writing this in a functional style, like:

```
const files = stdout.split('\n').reduce((line) =>  { <line processing logic>; files.push(file) }).filter(Boolean)
```

`filter(Boolean)` removes null/undefined from the the Array, which will be placed there if there is an early return in the `reduce` function.

Reply:
I'm having trouble finding how to select in TS in a functional style.  I wanted to do something like this, in one statement.
```
  async filesLocallyModified(): Promise<string[]> {
    let files: GitStatus[] = await this.gitStatus();
    return files
      .filter((file) => file.status[0] !== '?')
      .reduce((file) => file.file);
  }
```

Reply:
There are some examples here:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce

The function takes two arguments - the accumulator and the array entry

The second argument to the reduce function is the initial value - in this case, an empty Array.